### PR TITLE
Feature/add koa js

### PR DESCRIPTION
**Pull request by monostere0**
Added KoaJS as middleware

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -1,83 +1,78 @@
...
-  'angular2/bundles/angular2.dev.js',
-  'angular2/bundles/http.dev.js',
-  'angular2/bundles/router.dev.js'
+    //'es6-shim/es6-shim.js',
+    'systemjs/dist/system-polyfills.js',
+    'angular2/bundles/angular2-polyfills.js',
+    'systemjs/dist/system.src.js',
+    'rxjs/bundles/Rx.js',
+    'angular2/bundles/angular2.dev.js',
+    'angular2/bundles/http.dev.js'
```
Comment (adrianburcin) [inline Gulpfile.js]: @leudanielm 'angular2/bundles/router.dev.js' needs to be after http.dev.js

File: Gulpfile.js [hunk 2]
```diff
@@ -1,83 +1,78 @@
...
+        .pipe(gulp.dest(TARGET));
 });
 
 gulp.task('vendor-js', () => {
-  return gulp.src(VENDOR_FILES)
-    //.pipe($.uglify())
-    .pipe($.concatUtil('vendor.js'))
-    .pipe(gulp.dest(TARGET));
+    return gulp.src(VENDOR_FILES)
+        .pipe($.uglify())
```
Comment (adrianburcin) [inline Gulpfile.js]: @leudanielm Uglify will broke some angular2 providers

File: Gulpfile.js [hunk 3]
```diff
@@ -1,83 +1,78 @@
...
 gulp.task('app-css', () => {
-  return gulp.src(CSS_FILES)
-    .pipe($.sass().on('error', $.sass.logError))
-    .pipe($.autoprefixer())
-    //.pipe($.concatUtil('app.css'))
-    .pipe(gulp.dest(TARGET));
+    return gulp.src(CSS_FILES)
+        .pipe($.sass().on('error', $.sass.logError))
+        .pipe($.autoprefixer())
+        .pipe($.concatUtil('app.css'))
```
Comment (adrianburcin) [inline Gulpfile.js]: @leudanielm No need to concat and get all styles in the same file. It looks more natural to have them in styles folder and separate files, as angular 2 have dynamic styles loader (maybe will help)

File: Gulpfile.js [hunk 4]
```diff
@@ -1,83 +1,78 @@
...
-  gulp.watch(CSS_FILES, ['app-css', browserSync.reload]);
-  gulp.watch(HTML_FILES, ['app-html', browserSync.reload]);
+    gulp.watch(JS_FILES, ['app-js', browserSync.reload]);
+    gulp.watch(CSS_FILES, ['app-css', browserSync.reload]);
+    gulp.watch(HTML_FILES, ['app-html', browserSync.reload]);
 });
 
-gulp.task('default', $.sequence('app-html', 'vendor-js', 'app-js', 'app-css'));
\ No newline at end of file
+gulp.task('default', $.sequence('app-html', 'vendor-js', 'app-js', 'app-css', 'app-index'));
```
Comment (adrianburcin) [inline Gulpfile.js]: @leudanielm index.html was moved to src/, so app-index does nothing here

File: Gulpfile.js (hunk 1 above)
Comment (monostere0) [inline Gulpfile.js]: Fixed!

File: Gulpfile.js (hunk 2 above)
Comment (monostere0) [inline Gulpfile.js]: Fixed!

File: Gulpfile.js (hunk 3 above)
Comment (monostere0) [inline Gulpfile.js]: Fixed! Still on the old fashioned-way without SystemJS ;)

File: Gulpfile.js (hunk 4 above)
Comment (monostere0) [inline Gulpfile.js]: Fixed!

**Comment by adrianburcin** [comment]
:+1: :ship: it

**Pull request merged**